spyder_kernels/customize/spyderpdb.py

```diff
@@ -300,36 +308,73 @@ def interaction(self, frame, traceback):
             return super(SpyderPdb, self).interaction(
                 frame, traceback)
 
-    def print_stack_entry(self, frame_lineno, prompt_prefix='\n-> ',
-                          context=None):
+    def print_stack_entry(self, *args, **kwargs):
         """Disable printing stack entry if requested."""
         if self._disable_next_stack_entry:
             self._disable_next_stack_entry = False
             return
-        return super(SpyderPdb, self).print_stack_entry(
-            frame_lineno, prompt_prefix, context)
+        return super(SpyderPdb, self).print_stack_entry(*args, **kwargs)
 
     # --- Methods overriden for skipping libraries
     def stop_here(self, frame):
         """Check if pdb should stop here."""
-        if (frame is not None
-                and "__tracebackhide__" in frame.f_locals
-                and frame.f_locals["__tracebackhide__"] == "__pdb_exit__"):
+        # Never stop if we are continuing unless there is a breakpoint
+        if self.stopframe == self.botframe and self.stoplineno == -1:
```

Comment:
Optimisation: avoid calling the ipython mechanism if we know we won't stop anyway